build/commands/scripts/lit_mangler.ts

```diff
@@ -0,0 +1,200 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+// Note: This script runs as part of the build process, not at runtime, like our
+// old Polymer overrides used to. It generates a new .html.ts file with our
+// modifications to upstream's template.
+
+// Note: This file is stateful, and is meant to be executed once per mangled
+// file. It should be run via the command line.
+
+import ts from 'typescript'
+import { readFileSync, writeFileSync } from 'fs'
+import path from 'path'
+import { JSDOM } from 'jsdom'
+import commander from 'commander'
+
+export interface HTMLTemplateTags {
+    id: number,
+    text: string,
+    children: HTMLTemplateTags[]
+}
+
+let nextId = 1
+
+const loadRaw = (filepath: string) => {
+    const text = readFileSync(filepath, 'utf-8')
+    const tsFile = ts.createSourceFile(path.basename(filepath), text, ts.ScriptTarget.Latest)
+
+    return tsFile
+}
+
+const getTemplateLiterals = (source: ts.SourceFile, node: ts.Node, templateTags: HTMLTemplateTags) => {
+    if (ts.isTaggedTemplateExpression(node) && node.tag.getText(source) === 'html') {
+        let text = node.template.getText(source)
+        text = text.substring(1, text.length - 1)
+        const tag: HTMLTemplateTags = {
+            children: [],
+            text,
+            id: nextId++
+        }
+
+        getTemplateLiterals(source, node.template, tag)
+
+        templateTags.children.push(tag)
+        return
+    }
+
+    node.forEachChild(n => {
+        getTemplateLiterals(source, n, templateTags)
+    })
+}
+
+const getLiteralsFromFile = (filepath: string) => {
+    const file = loadRaw(filepath)
+    const root: HTMLTemplateTags = { text: file.getText(), children: [], id: nextId++ }
+    getTemplateLiterals(file, file, root)
+
+    if (root.children.length === 0) {
+        throw new Error(`No templates found in ${filepath}`)
+    }
+
+    return root
+}
+
+const injectPlaceholders = (template: HTMLTemplateTags) => {
```

Comment:
A couple of these internal functions might be a good place to add comments so a reader can understand how this works, e.g. why we need to add placeholders